web/src/components/network/IpAddressInput.tsx

```diff
@@ -20,17 +20,26 @@
  */
 
 import React, { useState } from "react";
-import { isValidIp } from "~/client/network/utils";
-import { TextInput, ValidatedOptions } from "@patternfly/react-core";
+import { isValidIp } from "~/utils/network";
+import { TextInput, TextInputProps, ValidatedOptions } from "@patternfly/react-core";
 import { _ } from "~/i18n";
 
-const IpAddressInput = ({ placeholder, onError = () => null, ...props }) => {
-  const [validated, setValidated] = useState("default");
+const IpAddressInput = ({
+  label = _("IP Address"),
+  onError = (value) => null,
+  ...props
+}: TextInputProps & { defaultValue?: string }) => {
+  const [validated, setValidated] = useState(
+    !props.defaultValue || props.defaultValue === "" || isValidIp(props.defaultValue)
```

Comment:
that complex ternary operator with extra complex condition really kills my eyes and brain...and I see it as getting code worse then in previous state. Is it possible to somehow improve it? like at least moving that condition outside of method call.

Reply:
Just a note: previous state was not working as expected when, although likely not possible, mounting the component with a wrong value as `defaultValue`.